### DISCUSSION TITLE
Allow selections and ctrl+f searches inside pdfs documents?

### DISCUSSION THREAD

**Discussion by brokedarius** [jellyfin/jellyfin#12207] [General]
Hey folks I think the title says it all in the cases of pdf files there are some small features like this that would be missing in comparison to the chrome or firefox pdf readers and I thought these would be nice to have?

Maybe some color inversion for people who prefer to read white on black would be nice too?

**Comment by FatherDivine** [comment]
Seems to not be a feature still, but would be nice. What we request is an Optical Character Recognition (OCR) plugin for Jellyfin. OCR is the technology Adobe Acrobat uses to make scanned documents and image-only PDFs searchable and editable.

**Reply by FatherDivine** [reply]
one thing that "might" be possible is using OCRmyPDF: A popular open-source command-line tool that adds a searchable text layer to your scanned PDF files. It uses the powerful Tesseract OCR engine and can also perform image processing like deskewing and cleaning.

Process: After installing it, you would run a command like ocrmypdf input.pdf output.pdf and then add output.pdf to your Jellyfin library.

Then the problem would be creating a plugin that can read this new PDF and, once detecting the text layer, can then allow the control+f or just menu item if on Android to search text. For now though, I'll just use Acrobat
